recipe/cmake.patch

```diff
@@ -0,0 +1,68 @@
+diff --git a/src/cpp/CMakeLists.txt b/src/cpp/CMakeLists.txt
+index 49c53f0..9f19406 100644
+--- a/src/cpp/CMakeLists.txt
++++ b/src/cpp/CMakeLists.txt
+@@ -28,25 +28,25 @@ if (BUILD_CUDA_LIB)
+     set_property(TARGET flann_cuda_s PROPERTY COMPILE_DEFINITIONS FLANN_STATIC)
+ endif()
+ 
+-if(CMAKE_SYSTEM_NAME STREQUAL "Linux" AND CMAKE_COMPILER_IS_GNUCC)
+-    add_library(flann_cpp SHARED "")
+-    set_target_properties(flann_cpp PROPERTIES LINKER_LANGUAGE CXX)
+-    target_link_libraries(flann_cpp -Wl,-whole-archive flann_cpp_s -Wl,-no-whole-archive)
+-
+-    if (BUILD_CUDA_LIB)
+-	    cuda_add_library(flann_cuda SHARED "")
+-        set_target_properties(flann_cuda PROPERTIES LINKER_LANGUAGE CXX)
+-        target_link_libraries(flann_cuda -Wl,-whole-archive flann_cuda_s -Wl,-no-whole-archive)
+-        set_property(TARGET flann_cpp_s PROPERTY COMPILE_DEFINITIONS FLANN_USE_CUDA)
+-    # 	target_link_libraries(flann_cuda cudpp_x86_64)
+-    endif()
+-else()
++#if(CMAKE_SYSTEM_NAME STREQUAL "Linux" AND CMAKE_COMPILER_IS_GNUCC)
```

Comment:
What's the value of commenting this on Windows? Based on the condition here it seems like this whole section would be skipped anyways and the `else` below run even on Windows.

Reply:
Note that this patch is applied on Linux, not on Windows.

I don't entirely know what the point is and it'd be good to clarify. :)

Reply:
As noted elsewhere, a patch is required to due to the way CMake 3.11 treats `add_library` commands with empty strings. That said, perhaps adding empty source files, as suggested at https://github.com/mariusmuja/flann/issues/369 may be a better workaround. Regardless, I'll document this issue more clearly in the meta.yaml.

Reply:
I see. So somewhere later on they are adding things to include in the construction of these libraries, but they need to initialize them first.

Reply:
Sorry must have overlooked the selector.

Reply:
Actually, I think they are creating the static library first and then trying to create the shared library directly from the static, but when CMake added this ability to add source files to the initialized (but empty) target, it messed up that logic.